frontend/packages/ceph-storage-plugin/src/components/kms-config/kms-config.tsx

```diff
@@ -66,10 +66,10 @@ export const KMSConfigure: React.FC<KMSConfigureProps> = ({
           id="kms-provider"
           name="kms-provider-name"
           aria-label={t('ceph-storage-plugin~kms-provider-name')}
-          isDisabled={!isOdf}
+          isDisabled={!isOdf || isLengthUnity(allowedKMSProviders)}
```

Comment:
need to add feature guard here instead of checking for `isOdf`, so that dropdown is disabled for ODF 4.9 + OCP 4.10 cases.